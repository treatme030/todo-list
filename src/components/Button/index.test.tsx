import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import "@testing-library/jest-dom/extend-expect";
import 'jest-styled-components';
import { Button } from './index';

describe('<Button />', () => {
  it('renders component correctly', () => {
    const { container } = render(<Button label="Button Test" />);

    const label = screen.getByText('Button Test');
    expect(label).toBeInTheDocument();

    const button = screen.getByRole('button');
    expect(button).toHaveStyleRule('background-color', '#304ffe');
    expect(button).toHaveStyleRule('background-color', '#1e40fe', {
      modifier: ':hover',
    });

    expect(container).toMatchSnapshot();
  });
  it('changes backgroundColor and hoverColor Props', () => {
    const backgroundColor = '#ff1744';
    const hoverColor = '#f01440';
    render(<Button label="Button Test" backgroundColor={backgroundColor} hoverColor={hoverColor} />);
    
    const button = screen.getByRole('button');
    expect(button).toHaveStyleRule('background-color', backgroundColor);
    expect(button).toHaveStyleRule('background-color', hoverColor, {
      modifier: ':hover',
    });
  });
  it('clicks the button', () => {
    const handleClick = jest.fn();
    render(<Button label="Button Test" onClick={handleClick} />);

    const label = screen.getByText('Button Test');
    expect(handleClick).toHaveBeenCalledTimes(0);
    userEvent.click(label);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});