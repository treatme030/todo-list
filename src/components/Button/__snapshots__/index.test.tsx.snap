// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders component correctly 1`] = `
.c0 {
  text-align: center;
  background-color: #304ffe;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
}

.c0:hover {
  background-color: #1e40fe;
}

.c0:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c1 {
  color: #fff;
  font-size: 16px;
}

<div>
  <button
    class="c0"
  >
    <label
      class="c1"
    >
      Button Test
    </label>
  </button>
</div>
`;
